### ISSUE TITLE
/usr/local/bin/gcc symlinked to /usr/local/bin/gcc-6

### ISSUE THREAD

**Issue by gilgameshskytrooper** [Homebrew/homebrew-core#18397]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

# Please always follow these steps:
- [👍 ] Confirmed this is a problem with `brew install`ing one, specific Homebrew/homebrew-core formula (not cask or tap) and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/caskroom/homebrew-cask/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [👍] Ran `brew update` and retried your prior step?
- [👍 ] Ran `brew doctor`, fixed all issues and retried your prior step? (had some random issues stemming from numpy and what not, but no issues with gcc
- [👍] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
```brew gist-logs gcc```
```Error: No logs.```
- [👍 ] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

```brew doctor```
```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/python3.5m/pygame/_camera.h
  /usr/local/include/python3.5m/pygame/_pygame.h
  /usr/local/include/python3.5m/pygame/_surface.h
  /usr/local/include/python3.5m/pygame/bitmask.h
  /usr/local/include/python3.5m/pygame/camera.h
  /usr/local/include/python3.5m/pygame/fastevents.h
  /usr/local/include/python3.5m/pygame/font.h
  /usr/local/include/python3.5m/pygame/freetype.h
  /usr/local/include/python3.5m/pygame/mask.h
  /usr/local/include/python3.5m/pygame/mixer.h
  /usr/local/include/python3.5m/pygame/pgarrinter.h
  /usr/local/include/python3.5m/pygame/pgbufferproxy.h
  /usr/local/include/python3.5m/pygame/pgcompat.h
  /usr/local/include/python3.5m/pygame/pgopengl.h
  /usr/local/include/python3.5m/pygame/pygame.h
  /usr/local/include/python3.5m/pygame/scrap.h
  /usr/local/include/python3.5m/pygame/surface.h
Warning: Use numpy instead of deprecated homebrew/python/numpy
Warning: Use numpy instead of deprecated homebrew/python/numpy
```
```brew config```
```
HOMEBREW_VERSION: 1.3.3-2-g7d720e8
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7d720e8fac6975aa402bf69361507982fbfc49ff
Last commit: 4 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 625b75eb11710dd77e67d11de728dcc110eb260c
Core tap last commit: 46 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Clang: 9.0 build 900
Git: 2.14.1 => /usr/local/bin/git
Perl: /usr/local/bin/perl => /usr/local/Cellar/perl/5.26.0/bin/perl
Python: /usr/local/opt/python/libexec/bin/python => /usr/local/Cellar/python/2.7.13_1/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /Users/andrewlee/.rvm/rubies/ruby-2.4.0/bin/ruby
Java: 1.8.0_102, 1.8.0_73, 1.8.0_66
macOS: 10.12.6-x86_64
Xcode: 9.0
CLT: N/A
X11: 2.7.11 => /opt/X11
```

To help us debug your issue please explain:
- What you were trying to do (and why)
I casually tried ```gcc --version``` because I learned gcc went from v.6.4 to v.7.2 in August. 
- What happened (include command output)
Surprisingly, gcc & g++ showed up as Homebrew GCC version 6.4.
```gcc --version```
```
gcc (Homebrew GCC 6.4.0) 6.4.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

However, when running ```brew install gcc```, I got ```Warning: gcc 7.2.0 is already installed``` which means brew internally sees the installed gcc version as 7.2.0 rather than 6.4.0

- What you expected to happen
Expected
```gcc --version```

```
gcc (Homebrew GCC 7.2.0) 7.2.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
- Step-by-step reproduction instructions (by running `brew install` commands)

None. Maybe fixed if I do a re-install

# Formula additions or changes
To get formulae added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
We will close issues requesting formulae changes.

Basically, when I did ```ls -l /usr/local/bin | grep gcc```, the entry for the ```/usr/local/bin/gcc``` symlink pointed to ```/usr/local/bin/gcc-6```

Hence, I just manually deleted the gcc and g++ symlinks, and made new ones that pointed to gcc-7 and g++-7

```rm /usr/local/bin/gcc```
```ln -s /usr/local/bin/gcc-7 /usr/local/bin/gcc```
```rm /usr/local/bin/g++```
```ln -s /usr/local/bin/g++-7 /usr/local/bin/g++```

Now everything is working fine. I looked through the homebrew-core/Formula/gcc.rb code, but I don't really see where the symlinking is happening. My guess is that the symlinking portion of the code just never got updated for 7.2.0 and keeps the gcc and g++ symlinked to the gcc/g++ 6.4.0

**Comment by ilovezfs** [comment]
What's the output of
```
brew unlink gcc
brew link gcc
```

**Comment by gilgameshskytrooper** [comment]
That didn't do anything. Essentially, it unliked gcc from gcc-6, but then ```brew link gcc``` relinked it back to gcc-6

**Comment by ilovezfs** [comment]
What's the output of
```
ls /usr/local/Cellar/gcc
```

**Comment by DomT4** [comment]
There shouldn't even be a vanilla `gcc` symlink to any Homebrew `gcc` formula, no? Because otherwise we're shadowing the system compiler.

**Comment by gilgameshskytrooper** [comment]
```ln /usr/local/Cellar/gcc```
```ln: /usr/local/Cellar/gcc: Is a directory```

**Comment by gilgameshskytrooper** [comment]
Every since I manually symlinked gcc 7.2.0, ```brew unlink gcc; and brew link gcc``` (on fish, it would be ```brew unlink gcc && brew link gcc``` on bash), it relinks to the correct version. Since I can no longer reproduce this problem, I will just close the issue. Will reopen if something similar happens in the future
